### ISSUE TITLE
Warning in HCIsInstanceOf

### ISSUE THREAD

**Issue by tomaszbak** [hamcrest/OCHamcrest#67]
Class *HCIsInstanceOf* has started to generate warning:
> Incompatible pointer types sending 'Class' to parameter of type 'NSString *'

for method:
```objc
+ (instancetype)isInstanceOf:(Class)type
{
    return [[self alloc] initWithType:type];
}
```
I'm using **XCode 6.3.1**

**Comment by scottrhoyt** [comment]
This seems to be because the compiler is looking at the method definition in CAEmitterBehavior.h instead of the definition from the superclass in HCClassMatcher.h.  Seeing as HCIsInstanceOf is not a child class of CAEmitterBehavior, I'm not sure why this is. Could be a compiler or Xcode issue?

**Comment by herzi** [comment]
I'm receiving the same warning in `HCIsTypeOf.m` as well.

Should I open a ticket at Apple's bug tracker?

**Comment by scottrhoyt** [comment]
I haven't dug deeper on this to figure out where the problem lies, so I cannot say whether it is an Apple bug or not. Still might be worth opening a ticket. Has anyone seen this error in the newest beta of Xcode?

**Comment by jtligon** [comment]
can confirm, still present in latest beta.
in addition, the new __nonnull warning is present as well.

HCIsInstanceOf.m:11:39: Incompatible pointer types sending 'Class' to parameter of type 'NSString * __nonnull'

**Comment by jonreid** [comment]
I didn't try Xcode 7 beta 1, but I don't see it when I build OCHamcrest in beta 2.

**Comment by herzi** [comment]
This issue is also fixed with the initial Xcode 7 beta. I think we can close this ticket now.

**Comment by jtligon** [comment]
Clean and build sent it away for Xcode 7 beta 1. thanks!

**Comment by jonreid** [comment]
Still happens with Xcode 6.4. Oh well, 7 is on the way.

**Comment by jonreid** [comment]
Fails on beta 4. :(

**Comment by herzi** [comment]
Just my two cents: To me, this clearly is a bug in Apple's toolchain. We should close this ticket and raise tickets at http://bugreport.apple.com/ to get this solved.

**Comment by jonreid** [comment]
OK, hold on: I don't see it in the Xcode 7 if I build OCHamcrest independently. I _do_ see it in Xcode 6.4 if it's being built as a Pod.

**Comment by jonreid** [comment]
Fixed in d7c0f55 by renaming the method. Apple could do better on this, since `[self alloc]` returns an instancetype.

**Comment by rein-achten** [comment]
@jonreid; Are you planning to officially release a version that incorporates this change?

**Comment by jonreid** [comment]
Just released v4.2.0

**Comment by rein-achten** [comment]
Thanks!
